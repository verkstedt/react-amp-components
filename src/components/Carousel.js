import React, { Component } from 'react'
import { string, oneOfType, arrayOf, node, bool, oneOf } from 'prop-types'
import { Helmet } from 'react-helmet'

import pick from '../utils/pick'
import commonPropTypes from './commonPropTypes'

class Carousel extends Component {
  static propTypes = {
    ...commonPropTypes,
    type: oneOf(['carousel', 'slides']).isRequired,
    autoplay: bool,
    delay: string,
    children: oneOfType([
      arrayOf(node),
      node
    ])
  }

  static defaultProps = {
    autoplay: null,
    children: [],
    delay: null
  }

  get properties() {
    return pick(this.props, ...[
      ...Object.keys(commonPropTypes),
      'type',
      'autoplay',
      'delay'
    ])
  }

  render() {
    const { children } = this.props
    return [
      <Helmet key="helmet">
        <script async="" custom-element="amp-carousel" src="https://cdn.ampproject.org/v0/amp-carousel-0.1.js" />
      </Helmet>,
      <amp-carousel {...{ ...this.properties }} key="carousel">
        {children}
      </amp-carousel>
    ]
  }
}

export default Carousel
