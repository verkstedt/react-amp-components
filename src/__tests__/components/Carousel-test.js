import React from 'react'
import Carousel from '../../components/Carousel'
import Helmet from '../../utils/Helmet'
import { renderComponent } from '../test-utils'

describe('Carousel', () => {
  it('matches the snapshot', () => {
    const carousel = renderComponent(
      <Carousel height="250" width="250" type="carousel">
        <p>Carousel contents</p>
      </Carousel>
    )
    expect(carousel.toJSON()).toMatchSnapshot()
  })

  it('injects the right script tag', () => {
    renderComponent(
      <Carousel height="250" width="250" type="carousel">
        <p>Carousel contents</p>
      </Carousel>
    )
    Helmet.canUseDOM = false
    const staticHead = Helmet.renderStatic()
    expect(staticHead.scriptTags).toMatchSnapshot()
  })

  it('passes down the right props', () => {
    const props = {
      height: '100',
      width: '100',
      type: 'carousel'
    }
    const carousel = renderComponent(
      <Carousel {...props}><p>content</p></Carousel>
    )
    expect(carousel.toJSON().props).toMatchObject(props)
  })
})
