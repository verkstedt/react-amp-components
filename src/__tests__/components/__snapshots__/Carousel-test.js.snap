// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel injects the right script tag 1`] = `
Array [
  Object {
    "async": "",
    "custom-element": "amp-carousel",
    "src": "https://cdn.ampproject.org/v0/amp-carousel-0.1.js",
  },
]
`;

exports[`Carousel matches the snapshot 1`] = `
<amp-carousel
  autoplay={null}
  delay={null}
  height="250"
  type="carousel"
  width="250"
>
  <p>
    Carousel contents
  </p>
</amp-carousel>
`;
